sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java

```diff
@@ -1203,6 +1207,22 @@ WithWindowsAdminUsernameManagedOrUnmanaged withLatestWindowsImage(
              * @return the next stage of the definition
              */
             WithCreate withOSDiskDiskEncryptionSet(String diskEncryptionSetId);
+
+            /**
+             * Specifies the OS disk to be ephemeral.
+             * @return the next stage of the definition
+             */
+            WithEphemeralOSDisk withEphemeralOSDisk();
+        }
+
+        /** The stage of a virtual machine definition allowing to select Ephemeral OS disk placement. */
+        interface WithEphemeralOSDisk {
+            /**
+             * Selects where you want to place the Ephemeral OS disk.
+             * @param placement placement of the Ephemeral OS disk
+             * @return the next stage of the definition
+             */
+            WithManagedCreate withPlacement(DiffDiskPlacement placement);
```

Comment:
So state transfer to `WithManagedCreate`, and user can only create managed data disk?

Reply:
Yes. Though document does not mention it specifically, evidences are:
1. In portal, unmanaged disks are not allowed: 
![image](https://user-images.githubusercontent.com/92354331/157444802-549abaf4-8f7c-47db-8d0d-5d5d82ee97a8.png)
2. In doc's Q&A session, they say that managed disks can be used along with ephemeral OS disk, they left out unmnaged disks.
3. Ephemeral OS disks are managed disks. Since managed disks cannot mix with unmanaged disks, I assume that unmanaged disks can not mix with ephemeral OS disk.